### ISSUE TITLE
[Feature: Add the option to enable cmp]

### ISSUE THREAD

**Issue by sergiornelas** [laytan/cloak.nvim#21]
Hello, amazing plugin btw.
Sometimes I want to use the cmp for snippets, or all kind of completions, doesn't matter if the hidden text shows in the cmp.
Would be amazing if you add an option to enable cmp:

```lua
require("cloak").setup({
        disable_cmp = false.  -- default true	
        -- ...
})
```
Thanks.

**Comment by whoop-t** [comment]
Could you just uncloak the file while you are working on it?

I see a possible use case being you are working on a file with cloaked stuff, but most parts of the file are not cloaked but wont have auto completion due to the whole file having cmp disabled.

**Comment by ph1losof** [comment]
I have implemented this feature in my plugin - [ecolog.nvim](https://github.com/philosofonusus/ecolog.nvim). Which provides all-in-one toolkit to work with environment variables and .env files

You can take a look at the docs section about this feature [here](https://github.com/philosofonusus/ecolog.nvim?tab=readme-ov-file#configuration-options-2)

And a comparison to `cloak.nvim` [here](https://github.com/philosofonusus/ecolog.nvim?tab=readme-ov-file#security-features-vs-cloaknvim)

**Comment by laytan** [comment]
So to clarify you want to enable cmp, but have the cmp results sourced from the buffer be cloaked? I would accept a good and small patch adding that, if even possible.

**Comment by ph1losof** [comment]
Yes, further more I think it will be appropriate to add `blink.cmp` support, didn't want to make an issue out of this small request. My plugin already handles it, I may create a PR to match it
